File: com-mini-core/src/main/java/com/mini/http/converter/StringConverter.java
package com.mini.http.converter;

import java.io.IOException;
import java.util.Objects;

import javax.annotation.Nonnull;

import okhttp3.Call;
import okhttp3.Response;
import okhttp3.ResponseBody;

public class StringConverter implements Converter<String> {
	public static final StringConverter INSTANCE = new StringConverter();

	@Override
	public String apply(@Nonnull Call call, @Nonnull Response response) throws IOException {
		if (!response.isSuccessful()) throw new IOException(response.message());
		try (ResponseBody body = Objects.requireNonNull(response.body())) {
			return body.string();
		}
	}

}

File: com-mini-kotlin/src/main/java/com/mini/core/http/converter/StringConverter.kt
package com.mini.core.http.converter

import okhttp3.Call
import okhttp3.Response
import java.io.IOException

@Suppress("unused")
class StringConverter : Converter<String?> {
    @Throws(IOException::class)
    override fun apply(call: Call, response: Response): String? {
        return if (response.isSuccessful) {
            response.body()?.use { body ->
                body.string()
            }
        } else throw IOException(response.message())
    }
}